File: 2022/Day_1/max_three_cal_counter.py
total_cals=[]
cals=0
with open("Day_1/cals.txt") as cal_file:
    for line in cal_file:
        if line.strip().isdigit():
            cals+=int(line)
        else:
            total_cals.append(cals)
            cals=0
print(sum(sorted(total_cals)[-3:]))

File: 2022/Day_1/CPP/max_three_cal_counter.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main()
{
    int cal {0};
    int total_cals {0};
    int cals[3];
    int max_cals={0};
    string line;
    ifstream InputFile("cals.txt");
    while (getline(InputFile, line))
    {
        if (sscanf(line.c_str(), "%d", &cal) == 1)
        {
            total_cals+=cal;
        }
        if (line.length() == 0)
        {
            max_cals=(max_cals < cals) ? cals: max_cals;
            total_cals=0;
        }
    }
    InputFile.close();
    cout<<max_cals<<endl;
}